Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -923,24 +923,34 @@ private class MacroApplication<Context: MacroExpansionContext>: SyntaxRewriter {
       return DeclSyntax(node)
     }
 
-    guard node.bindings.count == 1, let binding = node.bindings.first else {
+    guard node.bindings.count == 1 else {
       contextGenerator(Syntax(node)).addDiagnostics(
         from: MacroApplicationError.accessorMacroOnVariableWithMultipleBindings,
         node: node
       )
       return DeclSyntax(node)
     }
 
-    let expansion = expandAccessors(of: node, existingAccessors: binding.accessorBlock)
+    let bindingKeyPath = \VariableDeclSyntax.bindings[node.bindings.startIndex]
+    let binding = node[keyPath: bindingKeyPath]
+    var expansion = expandAccessors(of: node, existingAccessors: binding.accessorBlock)
+
     if expansion.accessors != binding.accessorBlock {
+      if binding.accessorBlock == nil {
+        // remove the trailing trivia of the variable declaration and move it
+        // to the trailing trivia of the left brace of the newly created accessor block
+        node[keyPath: bindingKeyPath].trailingTrivia = []
+        expansion.accessors?.leftBrace.trailingTrivia = binding.trailingTrivia
+      }
```

Comment:
This would still be an issue if the macro expands to an accessor that doesn’t have a newline after the `{`, right? Ie. you have a macro that expands to `{ 1 }` and has formatting disabled, so no newline gets added by `BasicFormat`.

If that’s the case, I would just move the trailing trivia after the closing brace. That way you can conceptually think of the accessor as being inserted before the binding’s trailing trivia.

Reply:
How could I disable formatting for a test case?

Reply:
You'll need to add a new macro and then set `static let formatMode: FormatMode = .disabled` in its definition. Also, nice catch @ahoppen! I forgot about that case :)

Reply:
I couldn't figure out how to make an `AccessorMacro` expand into something like `{ 1 }`, how could I do this?

Reply:
The following should test it

```swift
func testAccessorOnVariableDeclWithTrailingLineCommentAndAccessorBlock2() {
  struct ConstantOneSingleLineGetter: AccessorMacro {
    static var formatMode: FormatMode { .disabled }

    static func expansion(
      of node: AttributeSyntax,
      providingAccessorsOf declaration: some DeclSyntaxProtocol,
      in context: some MacroExpansionContext
    ) throws -> [AccessorDeclSyntax] {
      return [
        """
        get { 1 }
        """
      ]
    }
  }

  assertMacroExpansion(
    """
    @constantOne
    var x: Int // hello
    """,
    expandedSource: """
      var x: Int { // hello
        get { 1 }
      }
      """,
    macros: ["constantOne": ConstantOneSingleLineGetter.self],
    indentationWidth: indentationWidth
  )
}
```